src/pages/picture-upload/picture-upload.ts

```diff
@@ -30,27 +31,43 @@ export class PictureUploadPage {
     private alertCtrl: AlertController,
     private accountService: AccountService,
     private profileService: ProfileService,
-    private pictureUploadService: PictureUploadService
+    private pictureUploadService: PictureUploadService,
+    private loadingCtrl: LoadingController
   ) {}
 
   async ionViewDidLoad() {
-    this.storage.get('userId').then(userId => {
-      (async () => {
-        this.userId = userId;
-        this.account = await this.accountService.getAccount(this.userId);
-        this.profile = await this.profileService.getProfileByEmail(
-          this.account.email
-        );
-      })();
-    });
+    const profileId = this.navParams.get('id');
+    if (profileId) {
+      this.profile = await this.profileService.getProfileById(profileId);
+    } else {
+      this.storage.get('userId').then(userId => {
+        (async () => {
+          this.userId = userId;
+          this.account = await this.accountService.getAccount(this.userId);
+          this.profile = await this.profileService.getProfileByEmail(
+            this.account.email
+          );
+        })();
+      });
+    }
   }
 
-  uploadPicture() {
+  async uploadPicture() {
+    const loader = this.loadingCtrl.create();
+    loader.present();
     if (this.image) {
-      this.pictureUploadService.uploadPicture(this.image, this.profile);
+      await this.pictureUploadService.uploadPicture(this.image, this.profile);
+      loader.dismiss();
+      const alert = this.alertCtrl.create({
```

Comment:
I recommend changing this to a toast since it's a simple notification message that doesn't need user confirmation.